quickstart/201-k8s-cluster-with-tf-and-aks/main.tf

```diff
@@ -34,33 +38,38 @@ resource "azurerm_log_analytics_solution" "test" {
   }
 }
 
+resource "random_pet" "azurerm_kubernetes_cluster_name" {
+  prefix = "cluster"
+}
+
+resource "random_pet" "azurerm_kubernetes_cluster_dns_prefix" {
+  prefix = "dns"
+}
+
 resource "azurerm_kubernetes_cluster" "k8s" {
   location            = azurerm_resource_group.rg.location
-  name                = var.cluster_name
+  name                = random_pet.azurerm_kubernetes_cluster_name.id
   resource_group_name = azurerm_resource_group.rg.name
-  dns_prefix          = var.dns_prefix
-  tags                = {
-    Environment = "Development"
-  }
+  dns_prefix          = random_pet.azurerm_kubernetes_cluster_dns_prefix.id
 
   default_node_pool {
     name       = "agentpool"
     vm_size    = "Standard_D2_v2"
-    node_count = var.agent_count
+    node_count = var.node_count
   }
   linux_profile {
     admin_username = "ubuntu"
 
     ssh_key {
-      key_data = file(var.ssh_public_key)
+      key_data = jsondecode(azapi_resource_action.ssh_public_key_gen.output).publicKey
     }
   }
   network_profile {
     network_plugin    = "kubenet"
     load_balancer_sku = "standard"
   }
   service_principal {
-    client_id     = var.aks_service_principal_app_id
-    client_secret = var.aks_service_principal_client_secret
+    client_id     = azuread_service_principal.app.application_id
+    client_secret = azuread_service_principal_password.app.value
   }
-}
+}
```

Comment:
If we'd like to use `time_sleep` resource in this example, we need a `depends_on` here:

```hcl
depends_on = [time_sleep.wait_30_seconds]
```